### PR TITLE
[NFC] add explanation to register flags doc

### PR DESCRIPTION
**Pull request by fmayer**

### CODE REVIEW

File: llvm/docs/MIRLangRef.rst [hunk 1]
```diff
@@ -545,36 +545,47 @@ corresponding internal ``llvm::RegState`` representation:
...
      - ``RegState::Undef``
+     - Value of the register doesn't matter.
 
    * - ``internal``
      - ``RegState::InternalRead``
+     - Register reads a value that is defined inside the same instruction or bundle.
 
    * - ``early-clobber``
      - ``RegState::EarlyClobber``
+     - Register definition happens before uses.
```
Comment (arsenm) [inline llvm/docs/MIRLangRef.rst:580]: This one probably needs more elaboration

File: llvm/docs/MIRLangRef.rst [hunk 2]
```diff
@@ -545,36 +545,47 @@ corresponding internal ``llvm::RegState`` representation:
...
 
    * - ``debug-use``
      - ``RegState::Debug``
+     - Register 'use' is for debugging purpose.
 
    * - ``renamable``
      - ``RegState::Renamable``
+     - Register that may be renamed.
```
Comment (arsenm) [inline llvm/docs/MIRLangRef.rst:588]: Also probably needs more explanation

**Review by arsenm — APPROVED** [review]
Most of these could use more elaboration, especially earlyclobber and renamable

**Comment by fmayer** [comment]
> Most of these could use more elaboration, especially earlyclobber and renamable

These are just copy pasted from the enum values

**Comment by arsenm** [comment]
> > Most of these could use more elaboration, especially earlyclobber and renamable
> 
> These are just copy pasted from the enum value comments

Where I usually go to refer to them, and still don't understand what renamable actually does

File: llvm/docs/MIRLangRef.rst [hunk 3]
```diff
@@ -540,6 +540,9 @@ Register Flags
 The table below shows all of the possible register flags along with the
 corresponding internal ``llvm::RegState`` representation:
 
+..
+   Keep this in sync with MachineInstrBuilder.h
```
Comment (arsenm) [inline llvm/docs/MIRLangRef.rst:544]: Does this appear in the generated output?

Comment (fmayer) [reply llvm/docs/MIRLangRef.rst:544]: No: https://github.com/llvm/llvm-project/blob/6dab9ff72a1410989553285490f2a077c707c855/llvm/docs/MIRLangRef.rst

**Review by arsenm — APPROVED** [review]

**Pull request merged**